File: app/src/main/java/com/example/dars_17_advanced_rv/adapter/vh/NameVH.java
package com.example.dars_17_advanced_rv.adapter.vh;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.dars_17_advanced_rv.R;
import com.example.dars_17_advanced_rv.common.ItemClickListener;
import com.example.dars_17_advanced_rv.model.NameModel;

public class NameVH extends
        RecyclerView.ViewHolder
        implements View.OnClickListener {
    private TextView tv;
    private ItemClickListener listener;
    private NameModel model;

    public NameVH(@NonNull View view) {
        super(view);
        this.listener = (ItemClickListener) view.getContext();
        tv = view.findViewById(R.id.tv);
        view.setOnClickListener(this);
    }

    public void onBind(NameModel model) {
        this.model = model;
        tv.setText(model.getName());
    }

    @Override
    public void onClick(View v) {
        listener.onItemClick(model);
    }
}

File: app/src/main/java/com/example/dars_17_advanced_rv/adapter/vh/NameVH.kt
package com.example.dars_17_advanced_rv.adapter.vh

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.example.dars_17_advanced_rv.common.toast
import com.example.dars_17_advanced_rv.model.NameModel
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_name.*


class NameVH(override val containerView: View)
    : RecyclerView.ViewHolder(containerView), LayoutContainer {

    private lateinit var name: String

    init {
        containerView.setOnClickListener {
            if (::name.isInitialized)
                containerView.context.toast(name)
        }
    }

    fun onBind(model: NameModel) {
        name = model.name
        tv.text = model.name
    }
}